python/ray/experimental/channel/nccl_group.py

```diff
@@ -278,7 +281,10 @@ def allreduce(
         # TODO(wxdeng): Use check_async_error.
         self._cuda_stream.synchronize()
         if self._closed:
-            raise RayChannelError("NCCL group has been destroyed.")
+            raise RayChannelError(
+                "NCCL group has been destroyed. There may be a dtype mismatch "
+                "between different ranks."
```

Comment:
We can determine if `send_buf.dtype == recv_buf.dtype`? And print error message based on that?

Reply:
Typically, the assertion `send_buf.dtype == recv_buf.dtype` will never fail because RayCG derives the `shape` and `dtype` of `recv_buf` based on `send_buf`.

I added the error message here for cases where different actors within the same NCCL group use different dtype values for the allreduce operation. We have no way to check for this.

Reply:
> Typically, the assertion `send_buf.dtype == recv_buf.dtype` will never fail because RayCG derives the `shape` and `dtype` of `recv_buf` based on `send_buf`.
> 
> I added the error message here for cases where different actors within the same NCCL group use different `dtype` values for the allreduce operation. We have no way to check for this.

Reply:
"between input tensors from different ranks"

Reply:
`NCCL group has been destroyed unexpectedly in allreduce operation. There may be a dtype mismatch between input tensors from different ranks.`

Reply:
updated: https://github.com/ray-project/ray/pull/49300/commits/11193a6cbf3dad404662180a4ec7fe08a73253a3